File: src/test/java/com/technews/scheduler/BlogJavaInsideSchedulerTest.java
package com.technews.scheduler;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;

@Disabled
@ActiveProfiles("local")
@SpringBootTest
class BlogJavaInsideSchedulerTest {

    @Autowired
    private BlogJavaInsideScheduler blogJavaInsideScheduler;

    @Test
    void run() {
        blogJavaInsideScheduler.runScheduled();
    }
    
    @Test 
    void getPostInfo() {
        final String postInfo = "Joakim Nordström on April 12, 2024";
        final String[] split = postInfo.split(" on ");

        Assertions.assertEquals("Joakim Nordström", split[0]);
        Assertions.assertEquals("April 12, 2024", split[1]);
    }

    @Test
    void getPostInfo_2() {
        final String postInfo = "April 12, 2024";
        final String[] split = postInfo.split(" on ");

        Assertions.assertEquals("April 12, 2024", split[0]);
    }
}

File: src/test/java/com/technews/scheduler/BlogJavaInsideSchedulerTest.kt
package com.technews.scheduler

import com.technews.IntegrationTest
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import org.springframework.beans.factory.annotation.Autowired

@IntegrationTest
class BlogJavaInsideSchedulerTest(
    @Autowired private val blogJavaInsideScheduler: BlogJavaInsideScheduler
) : StringSpec({
    "Inside Java 게시물 스크래핑 동작" {
        blogJavaInsideScheduler.runScheduled()
    }

    "게시물 작성자와 날짜 파싱" {
        val postInfo = "Joakim Nordström on April 12, 2024"
        val split = postInfo.split(" on ")

        split[0] shouldBe "Joakim Nordström"
        split[1] shouldBe "April 12, 2024"
    }

    "게시물 날짜 파싱" {
        val postInfo = "April 12, 2024"
        val split = postInfo.split(" on ")

        split[0] shouldBe "April 12, 2024"
    }
})
